pkg/bloomgateway/processor.go

```diff
@@ -0,0 +1,169 @@
+package bloomgateway
+
+import (
+	"context"
+	"math"
+	"sort"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/common/model"
+
+	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/bloomshipper"
+)
+
+type tasksForBlock struct {
+	blockRef bloomshipper.BlockRef
+	tasks    []Task
+}
+
+type metaLoader interface {
+	LoadMetas(context.Context, bloomshipper.MetaSearchParams) ([]bloomshipper.Meta, error)
+}
+
+type blockLoader interface {
+	LoadBlocks(context.Context, []bloomshipper.BlockRef) (v1.Iterator[bloomshipper.BlockQuerierWithFingerprintRange], error)
+}
+
+type store interface {
+	blockLoader
+	metaLoader
+}
+
+type processor struct {
+	store  store
+	logger log.Logger
+}
+
+func (p *processor) run(ctx context.Context, tasks []Task) error {
```

Comment:
nit: I feel like `run`, `do` and `process` not descriptive enough names and to me they mean pretty much the same.

Reply:
Good point. I will give them more descriptive names.